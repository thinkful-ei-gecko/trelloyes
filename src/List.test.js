import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('test list component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <List key={2} 
      header='header string' 
      cards={[{id: 1, title: 'title 1', content: 'content 1'},
      {id: 2, title: 'title 2', content: 'content 2'}]} 
    />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const tree = renderer
    .create(
    <List key={2} 
      header='header string' 
      cards={[{id: 1, title: 'title 1', content: 'content 1'},
      {id: 2, title: 'title 2', content: 'content 2'}]} 
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
})