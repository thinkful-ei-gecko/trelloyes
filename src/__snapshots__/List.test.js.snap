// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test list component should match snapshot 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header string
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        title 1
      </h3>
      <p>
        content 1
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        title 2
      </h3>
      <p>
        content 2
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
